### DISCUSSION TITLE
Cannot create a session after oauth with google using vue3 todo example

### DISCUSSION THREAD

**Discussion by designlook** [supabase/supabase#5973] [General]
I am using most of code from Vue3 example at https://github.com/supabase/supabase/tree/master/examples/vue3-ts-todo-list

I am running it on localhost:3000. My supabase is running on docker at mysite.com:3000 and auth is at mysite.com:8001

Following is what my supabase auth configuration is:

```
{
    "supabaseUrl": "http://mysite.com:8001/",
    "supabaseKey": "eyxxxxxxxxxxxxxxxxxxx",
    "restUrl": "http://mysite.com:8001/rest/v1",
    "realtimeUrl": "ws://mysite.com:8001/realtime/v1",
    "authUrl": "http://mysite.com:8001/auth/v1",
    "storageUrl": "http://mysite.com:8001/storage/v1",
    "schema": "public",
    "multiTab": true,
    "headers": {
        "X-Client-Info": "supabase-js/1.30.3"
    },
    "auth": {
        "stateChangeEmitters": {},
        "currentUser": null,
        "currentSession": null,
        "autoRefreshToken": true,
        "persistSession": true,
        "multiTab": true,
        "localStorage": {},
        "api": {
            "url": "http://mysite.com:8001/auth/v1",
            "headers": {
                "X-Client-Info": "supabase-js/1.30.3",
                "Authorization": "Bearer eyxxxxxxxxxxxxxxxxxxx",
                "apikey": "eyxxxxxxxxxxxxxxxxxxx"
            },
            "cookieOptions": {
                "name": "sb",
                "lifetime": 28800,
                "domain": "",
                "path": "/",
                "sameSite": "lax"
            }
        }
    },
}
```

After I log in using Google account, it redirects me to http://localhost:3000?state=eyxxxxxxxxxxxxxxxxxxxxxxxxxxxx

But no user session is created. User is null.... I have the following code in main.js but it's never triggered.

```
supabase.auth.onAuthStateChange((event, session) => {
  userSession.value = session
})

```

Anyone have any ideas?

**Comment by awalias** [comment]
sounds like gotrue has some of it's default values set in the docker file or `.env` file, which you'll need to update and restart the container, see:

https://github.com/supabase/supabase/blob/6344481e088c45ae44c0ea41f5890dc1dd7b4ea5/docker/.env.example#L12
https://github.com/supabase/supabase/blob/6344481e088c45ae44c0ea41f5890dc1dd7b4ea5/docker/docker-compose.yml#L54

**Reply by designlook** [reply]
The solution that finally got it working was updating 

GOTRUE_EXTERNAL_GOOGLE_REDIRECT_URI=http://mysite.com:8001/auth/v1/callback

I had to add '/auth/v1/callback' to the end. Then the example app worked. Wonder if I missed that in the documentation anywhere. But dont recall any doc mentioning the auth/v1/callback is required.

Update: I did find some info at https://supabase.com/docs/guides/auth/auth-google where it mentioned the /auth/v1/callback required for Google Login.
